hoomd/data/syncedlist.py

```diff
@@ -212,6 +213,20 @@ def _validate_or_error(self, value):
             raise ValueError(f"Validation failed: {verr.args[0]}") from verr
         return processed_value
 
+    @property
+    def _simulation(self):
+        sim = self._simulation_
+        if sim is not None:
+            sim = sim()
+            if sim is not None:
+                return sim
```

Comment:
What should this property return in the case that `sim is None` after dereferencing the weakref to `sim`? I can't imagine code that uses the `_simulation` property will want to always have to double check to make sure `_simulation` actually returned something.

Reply:
As far as I understand it, that case should never happen. All objects holding this are held by a `Simulation`. `_simulation` is a back reference to that object. When `_simulation` returns None (the implied return value when there is no `return` statement), it is the same as if the object has not yet been added to a Simulation.

We could add this check for completeness. The only way to trigger it would be to directly access internal data structures and bypass HOOMD's attach/detach and/or add/remove APIs.
```python
            if sim is not None:
                return sim
            else:
                raise RuntimeError("Parent Simulation was garbage collected.")
```

Reply:
Yeah, we could error here or make sure we are detached (which would be my preference. This could happen in some unusual cases where the operation's lifespan is longer than the simulation's.

Reply:
In my original comment, I didn't think about the implied return value of `None` when there is no return statement, but I like the idea of fully checking for attachment when the weakref points to `None`